Add project repository url

sonlinux opened the pull request:
Fix #634 

Added project repository url to the project model, form and project detail page above the project url.

![inasafe](https://user-images.githubusercontent.com/10270148/36467916-af278abe-16e9-11e8-9f96-01288b6debc1.png)

timlinux commented on django_project/base/migrations/0019_project_project_repository_url.py:
```diff
@@ -0,0 +1,19 @@
+        migrations.AddField(
+            model_name='project',
+            name='project_repository_url',
+            field=models.URLField(help_text='A repository url for this project i.e github repository url', null=True, blank=True),
```
GitHub

timlinux replied:
e.g. not i.e.

sonlinux replied:
Alright @timlinux thanks
Updated.

timlinux replied:
You still have the old text....

```field=models.URLField(help_text='A repository url for this project i.e github repository url', null=True, blank=True),```

becomes

```
field=models.URLField(help_text='A repository url for this project e.g. GitHub repository url.', null=True, blank=True),
```

timlinux commented on django_project/base/models/project.py:
```diff
@@ -158,6 +158,13 @@ class Project(models.Model):
 
+    project_repository_url = models.URLField(
+        help_text=u'A repository url for this project '
+                  u'i.e github repository url',
```
e.g. GitHub

Also please always use complete sentances (with full stop terminating them)

sonlinux replied:
Alright @timlinux thanks
Updated.

timlinux commented on django_project/base/models/project.py:
```diff
@@ -158,6 +158,13 @@ class Project(models.Model):
 
+    project_repository_url = models.URLField(
+        help_text=u'A repository url for this project '
+                  u'e.g a github repository url.',
```
Please check carefully my comment:

```
      help_text=u'A repository url for this project '
                 u'e.g a github repository url.',
```

should be

```
      help_text=u'A repository URL for this project '
                 u'e.g, a GitHub repository URL.',
```

Also your string contruction is strange, it should be:

```
      help_text=(
          u'A repository url for this project '
          'e.g. a GitHub repository URL.'),
```

timlinux commented:
@sonlinux from our discussion this would come into code by proxy class - since I am now resorting to giving you exactly the code you needed to write.

sonlinux commented:
Ah sorry about that @timlinux 
I corrected then reset the commit because I wanted to just add 'for example' but I commited before that change.
Ill pay more attention.
I have updated

timlinux commented:
Not all the code is updated - also use URL rather than url in your text.

Gustry commented:
Maybe you should display the link somewhere in a template?

For the migration, remove the old migration file you created in this PR and re-run the migration command. Or just re-run the make update migration command to add another migration file

Gustry commented on django_project/base/models/project.py:
```diff
@@ -159,8 +159,8 @@ class Project(models.Model):
-        help_text=u'A repository URL for this project '
-                  u'e.g, a Github repository URL.',
+        help_text=u'A repository URL for this project. '
+                  u'For instance a path to the project\s Github repository.',
```
As Tim mentioned, better to put bracket when you are doing multiline strings.
```
help_text=(
    u'Looooong string'),
```
Example: https://github.com/inasafe/inasafe/blob/develop/safe/definitions/layer_purposes.py#L86

Also `GitHub` not `github`: https://en.wikipedia.org/wiki/GitHub 
And also the apostrophe after project?

sonlinux commented:
Thanks @Gustry 
I have displayed the link above project home url with an icon.
And for the migrations I have rerun and they are reflecting.

timlinux commented on django_project/base/models/project.py:
```diff
@@ -158,6 +158,13 @@ class Project(models.Model):
 
+    project_repository_url = models.URLField(
+        help_text=u'A repository URL for this project. '
+                  u'For instance a path to the project\'s Github repository.',
```
Github -> GitHub

sonlinux replied:
:-) Okay @timlinux 
Thanks

timlinux commented on django_project/base/migrations/0019_project_project_repository_url.py:
```diff
@@ -0,0 +1,19 @@
+        migrations.AddField(
+            model_name='project',
+            name='project_repository_url',
+            field=models.URLField(help_text="A repository URL for this project. For instance a path to the project's GitHub repository.", null=True, blank=True),
```
PEP8

Gustry replied:
This is a migration file. They are autogenerated by django so excluded from the PEP8 test in Travis and makefile.

timlinux commented:
LGTM thanks
